lvsfunc/aa.py

```diff
@@ -14,64 +14,80 @@
-
     Original function written by Zastin, modified by LightArrowsEXE.
-
     Dependencies:
-
     * kagefunc (optional: retinex edgemask)
     * vapoursynth-retinex (optional: retinex edgemask)
     * vapoursynth-tcanny (optional: retinex edgemask)
     * vapoursynth-eedi3
-    * vapoursynth-nnedi3 or znedi3
+    * vapoursynth-nnedi3
     * vapoursynth-nnedi3cl (optional: opencl)
-    * vsTAAmbk
-
     :param clip:                Input clip
     :param strength:            Set threshold strength (Default: 1)
     :param mask:                Clip to use for custom mask (Default: None)
     :param ret_mask:            Replace default mask with a retinex edgemask (Default: False)
     :param show_mask:           Return mask instead of clip (Default: False)
     :param opencl:              OpenCL acceleration (Default: False)
-
+    :param nnedi3cl:            OpenCL acceleration for nnedi3 (Default: False)
+    :param eedi3cl:             OpenCL acceleration for eedi3 (Default: False)
     :return:                    Antialiased clip
     """
-    try:
-        from vsTAAmbk import TAAmbk
-    except ModuleNotFoundError:
-        raise ModuleNotFoundError("nnedi3_clamp: missing dependency 'vsTAAmbk'")
-
+    
     if clip.format is None:
         raise ValueError("nneedi3_clamp: 'Variable-format clips not supported'")
-
+    
+    clip_y = get_y(clip)
+    
     bits = clip.format.bits_per_sample - 8
-    thr = strength * (1 >> bits)
-    strong = TAAmbk(clip, aatype='Eedi3', alpha=0.25, beta=0.5, gamma=40, nrad=2, mdis=20, mtype=0, opencl=opencl)
-    weak = TAAmbk(clip, aatype='Nnedi3', nsize=3, nns=3, qual=1, mtype=0, opencl=opencl)
-    expr = 'x z - y z - * 0 < y x y {0} + min y {0} - max ?'.format(thr)
-
-    if clip.format.num_planes > 1:
-        aa = core.std.Expr([strong, weak, clip], [expr, ''])
-    else:
-        aa = core.std.Expr([strong, weak, clip], expr)
-
-    if mask:
-        merged = clip.std.MaskedMerge(aa, mask, planes=0)
-    elif ret_mask:
-        try:
-            import kagefunc as kgf
-        except ModuleNotFoundError:
-            raise ModuleNotFoundError("nnedi3_clamp: missing dependency 'kagefunc'")
-        mask = kgf.retinex_edgemask(clip, 1).std.Binarize()
-        merged = clip.std.MaskedMerge(aa, mask, planes=0)
+    thr = strength * (1 >> bits) if clip.format.sample_type==vs.INTEGER else strength/219
+    expr = 'x z - y z - xor y x y {0} + min y {0} - max ?'.format(thr)
+    
+    if mask is None:
+        if ret_mask:
+            try:
+                import kagefunc as kgf
+            except ModuleNotFoundError:
+                raise ModuleNotFoundError("nnedi3_clamp: missing dependency 'kagefunc'")
+            mask = kgf.retinex_edgemask(clip_y, 1).std.Binarize()
+        else:
+            mask = clip_y.std.Prewitt().std.Binarize().std.Maximum().std.Convolution([1] * 9)
+    
+    nnedi3cl = fallback(nnedi3cl, opencl)
+    eedi3cl = fallback(eedi3cl, opencl)
```

Comment:
Is there really a reason to use a fallback here as opposed to just getting rid of the `opencl` parameter altogether?